storage/innobase/handler/ha_innodb.cc

```diff
@@ -4024,7 +4024,7 @@ static int innodb_init(void* p)
 			    test_filename));
 #endif /* DBUG_OFF */
 
-	os_file_set_umask(my_umask);
+	os_file_set_umask((mode_t) my_umask);
```

Comment:
What if we just remove this entire function as well as `os_innodb_umask`, and change the type of `my_umask` to `mode_t`?

I believe that the reason for this kind of data structure duplication is that originally, the intention was to embed InnoDB in multiple different databases, or to develop it as a stand-alone product outside MySQL.

Reply:
Ype, can remove function and go straight though.

The definition of `my_umask` looks like an exposed ABI so I was avoiding changing it:
`include/my_sys.h:extern MYSQL_PLUGIN_IMPORT int my_umask;               /* Default creation mask  */`. 

From the clang warnings this is only a sign difference as opposed to innodb that has the type longer.

Reply:
Ack, updated to remove innodb implementation and normalize the mode_t types elsewhere in the codebase.